src/Adyen/Service/ResourceModel/Payment/TerminalCloudAPI.php

```diff
@@ -7,13 +7,13 @@ class TerminalCloudAPI extends \Adyen\Service\AbstractResource
 
     protected $_requiredFields = array(
         'SaleToPOIRequest.MessageHeader.POIID',
-        'SaleToPOIRequest.MessageHeader.ServiceID',
-        //TODO fix: actually works with only first two levels
-        'SaleToPOIRequest.PaymentRequest.SaleData.SaleTransactionID.TransactionID', //reference
-        'SaleToPOIRequest.PaymentRequest.PaymentTransaction.AmountsReq.Currency',
-        'SaleToPOIRequest.PaymentRequest.PaymentTransaction.AmountsReq.RequestedAmount',
-        //PaymentData is optional, if not provided it will perform an authorisation(no refunds)
-        'SaleToPOIRequest.PaymentRequest.PaymentData.PaymentType',
+        'SaleToPOIRequest.MessageHeader.ServiceID'
+        // /sync can be a status call, in that case we don't send PaymentRequest
+//        'SaleToPOIRequest.PaymentRequest.SaleData.SaleTransactionID.TransactionID', //reference
+//        'SaleToPOIRequest.PaymentRequest.PaymentTransaction.AmountsReq.Currency',
+//        'SaleToPOIRequest.PaymentRequest.PaymentTransaction.AmountsReq.RequestedAmount',
+//        //PaymentData is optional, if not provided it will perform an authorisation(no refunds)
+//        'SaleToPOIRequest.PaymentRequest.PaymentData.PaymentType',
     );
```

Comment:
Can we just remove the complete requiredFields ?

Reply:
removing all the required fields in a separate PR